xmppserver/src/main/java/org/jivesoftware/openfire/commands/admin/EditBlockedList.java

```diff
@@ -0,0 +1,167 @@
+                    requestError = true;
+                }
+
+                listed.add(domain.getDomain());
+            }
+            catch ( NullPointerException npe )
+            {
+                note.addAttribute( "type", "error" );
+                note.setText(LocaleUtils.getLocalizedString("commands.admin.editblockedlist.note.jid-required", preferredLocale));
+                requestError = true;
+            }
+            catch (IllegalArgumentException npe)
+            {
+                note.addAttribute( "type", "error" );
+                note.setText(LocaleUtils.getLocalizedString("commands.admin.editblockedlist.note.jid-invalid", preferredLocale));
+                requestError = true;
+            }
+        }
+
+        if ( requestError )
+        {
+            // We've collected all errors. Return without applying changes.
+            return;
+        }
+
+        // No errors.
+        final Collection<RemoteServerConfiguration> blockedServers = RemoteServerManager.getBlockedServers();
+        for (final String domain : listed) {
+            if (blockedServers.stream().noneMatch(s -> s.getDomain().equals(domain))) {
+                RemoteServerManager.blockAccess(domain);
+            }
+        }
+        for (final RemoteServerConfiguration blockedServer : blockedServers) {
+            if ( !listed.contains(blockedServer.getDomain())) {
+                // Unblock - no longer on the list.
+                RemoteServerManager.deleteConfiguration(blockedServer.getDomain());
+            }
+        }
+
+        // Answer that the operation was successful
+        note.addAttribute("type", "info");
+        note.setText(LocaleUtils.getLocalizedString("commands.global.operation.finished.success", preferredLocale));
+    }
+
+    @Override
+    protected void addStageInformation(@Nonnull final SessionData data, Element command) {
+        final Locale preferredLocale = SessionManager.getInstance().getLocaleForSession(data.getOwner());
+
+        DataForm form = new DataForm(DataForm.Type.form);
+        form.setTitle(LocaleUtils.getLocalizedString("commands.admin.editblockedlist.form.title", preferredLocale));
+        form.addInstruction(LocaleUtils.getLocalizedString("commands.admin.editblockedlist.form.instruction", preferredLocale));
+
+        FormField field = form.addField();
+        field.setType(FormField.Type.hidden);
+        field.setVariable("FORM_TYPE");
+        field.addValue("http://jabber.org/protocol/admin");
+
+        field = form.addField();
+        field.setType(FormField.Type.jid_multi);
+        field.setLabel(LocaleUtils.getLocalizedString("commands.admin.editblockedlist.form.field.blacklistjids.label", preferredLocale));
+        field.setVariable("blacklistjids");
+        field.setRequired(true);
+        for (RemoteServerConfiguration blockedServer : RemoteServerManager.getBlockedServers()) {
+            field.addValue(new JID(null, blockedServer.getDomain(), null));
+        }
```

Comment:
There's no way using this form to return the blacklist to an empty list. Some of that might just be a Smack implementation detail about dealing with valueless lists, but all the same, we could deal with this in Openfire by either amending this command, or adding a command to wipe the blacklist.

Reply:
For consistency, I'd prefer this to be fixed in the client. An empty list is what we'd expect to have here. I fear that by allowing for that, but also providing a different means of clearing the list will only introduce confusion in the long run.

Reply:
Am working with Flow on a solution.

I say working with Flow. More like, I've handed him the problem with sufficient repro steps :)